src/Persistence_matrix/include/gudhi/Persistence_matrix/chain_matrix.h

```diff
@@ -0,0 +1,1351 @@
+   * 
+   * @param columnIndex @ref MatIdx index of the column
+   * @return The @ref rowindex "row index" of the pivot.
+   */
+  id_index get_pivot(index columnIndex);
+
+  /**
+   * @brief Resets the matrix to an empty matrix.
+   * 
+   * @param colSettings Pointer to an existing setting structure for the columns. The structure should contain all
+   * the necessary external classes specifically necessary for the choosen column type, such as custom allocators.
+   */
+  void reset(Column_settings* colSettings) {
+    matrix_.clear();
+    pivotToColumnIndex_.clear();
+    nextIndex_ = 0;
+    colSettings_ = colSettings;
+  }
+
+  /**
+   * @brief Assign operator.
+   */
+  Chain_matrix& operator=(const Chain_matrix& other);
+  /**
+   * @brief Swap operator.
+   */
+  friend void swap(Chain_matrix& matrix1, Chain_matrix& matrix2) {
+    swap(static_cast<typename Master_matrix::Matrix_dimension_option&>(matrix1),
+         static_cast<typename Master_matrix::Matrix_dimension_option&>(matrix2));
+    swap(static_cast<typename Master_matrix::Chain_pairing_option&>(matrix1),
+         static_cast<typename Master_matrix::Chain_pairing_option&>(matrix2));
+    swap(static_cast<typename Master_matrix::Chain_vine_swap_option&>(matrix1),
+         static_cast<typename Master_matrix::Chain_vine_swap_option&>(matrix2));
+    swap(static_cast<typename Master_matrix::Chain_representative_cycles_option&>(matrix1),
+         static_cast<typename Master_matrix::Chain_representative_cycles_option&>(matrix2));
+    matrix1.matrix_.swap(matrix2.matrix_);
+    matrix1.pivotToColumnIndex_.swap(matrix2.pivotToColumnIndex_);
+    std::swap(matrix1.nextIndex_, matrix2.nextIndex_);
+    std::swap(matrix1.colSettings_, matrix2.colSettings_);
+
+    if constexpr (Master_matrix::Option_list::has_row_access) {
+      swap(static_cast<typename Master_matrix::Matrix_row_access_option&>(matrix1),
+           static_cast<typename Master_matrix::Matrix_row_access_option&>(matrix2));
+    }
+  }
+
+  void print() const;  // for debug
+
+  friend class Id_to_index_overlay<Chain_matrix<Master_matrix>, Master_matrix>;
+
+ private:
+  using dim_opt = typename Master_matrix::Matrix_dimension_option;
+  using swap_opt = typename Master_matrix::Chain_vine_swap_option;
+  using pair_opt = typename Master_matrix::Chain_pairing_option;
+  using rep_opt = typename Master_matrix::Chain_representative_cycles_option;
+  using ra_opt = typename Master_matrix::Matrix_row_access_option;
+  using matrix_type = typename Master_matrix::column_container_type;
+  using dictionnary_type = typename Master_matrix::template dictionnary_type<index>;
+  using barcode_type = typename Master_matrix::barcode_type;
+  using bar_dictionnary_type = typename Master_matrix::bar_dictionnary_type;
+  using tmp_column_type = typename std::conditional<
+      Master_matrix::Option_list::is_z2, 
+      std::set<id_index>,
+      std::map<id_index, Field_element_type>
+    >::type;
```

Comment:
It is surprising that this is not one of the types defined in Persistence_matrix/columns.